thriftclient/client_pool.go

```diff
@@ -40,6 +40,10 @@ type ClientPoolConfig struct {
 	// SocketTimeout is the timeout on the underling thrift.TSocket.
 	SocketTimeout time.Duration
 
+	// Any labels that should be applied to metrics logged by the ClientPool.
+	// This includes the optional pool stats.
+	MetricsLabels []string
```

Comment:
Worth mention that it should be in even number in key value pairs, for example: `key1, value1, key2, value2`. The underlying implementation will actually panic if it's odd number.

Reply:
Alternative is that we accept a map here and convert that into string slice in constructors. (nil map will still work)

Reply:
I have a slightly different approach that I just pushed.